sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -950,21 +950,31 @@ CURLcode CurlSession::ReadStatusLineAndHeadersFromRawResponse(
     // The value can also be "keep-alive, close" (i.e. "both are fine"), in which case we are
     // preferring to treat it as keep-alive.
     // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection)
-    // Should it come to HTTP/2 and HTTP/3, they are "keep-alive", but any response from HTTP/2 or
-    // /3 containing a "Connection" header should be considered malformed. (HTTP/2:
-    // https://httpwg.org/specs/rfc9113.html#ConnectionSpecific
-    //  HTTP/3: https://httpwg.org/specs/rfc9114.html#rfc.section.4.2)
-    if (m_response->GetMajorVersion() == 1 && m_response->GetMinorVersion() >= 1)
+    if (m_response->GetMajorVersion() == 1)
     {
-      m_httpKeepAlive = (!hasConnectionClose || hasConnectionKeepAlive);
-    }
-    else if (m_response->GetMajorVersion() <= 1)
-    {
-      m_httpKeepAlive = hasConnectionKeepAlive;
+      if (m_response->GetMinorVersion() >= 1)
+      {
+        // HTTP/1.1+
+        m_httpKeepAlive = (!hasConnectionClose || hasConnectionKeepAlive);
+      }
+      else
+      {
+        // HTTP/1.0
+        m_httpKeepAlive = hasConnectionKeepAlive;
+      }
     }
     else
     {
-      m_httpKeepAlive = true;
+      Log::Write(
+          Logger::Level::Verbose,
+          LogMsgPrefix
+              + "Unsupported HTTP version in the response. Only HTTP/1.0 and HTTP/1.1 is "
+                "supported as a response, for an HTTP/1.1 request.");
+
+      // We don't expect HTTP/2.0 or 3.0 as a response, when sending an HTTP/1.1 request.
+      // Barring rejecting as malformed, the safest thing to do here is to assume the connection is
+      // not reusable.
+      m_httpKeepAlive = false;
```

Comment:
For reviewers: This is the primary change here that impacts the semantics. We are marking `httpKeepAlive` as false instead of true when we hit a case that is **not** HTTP 1.x, which is not supported by the SDK anyway.

Reply:
@Jinming-Hu can you please review this PR, as an extra set of eyes. It doesn't change any supported scenarios for storage (since we only care about HTTP/1.1 there), and I don't expect it to make any perf impact.

I am requesting your review purely out of abundance of caution :)